### PR TITLE
refactor(rolling_upgrade): logging and faster pipeline exit

### PR DESCRIPTION
**Pull request by KnifeyMoloko**
This PR is a port of the changes introduced in master with: https://github.com/scylladb/scylla-cluster-tests/pull/4603

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ]  ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/utils/version_utils.py [hunk 1]
```diff
@@ -192,6 +219,18 @@ def get_branch_version(url):
...
+def get_all_versions(url: str) -> set[str]:
+    repo_details = get_repository_details(url=url)
+    urls = get_scylla_urls_from_repository(repo_details=repo_details)
+
+    if repo_details.type == ScyllaFileType.DEBIAN:
+        return get_all_versions_from_debian_repository(urls=urls)
+    elif repo_details.type == ScyllaFileType.YUM:
+        return get_all_versions_from_centos_repository(urls=urls)
+    # To overcome on Pylint's "inconsistent-return-statements", a value must be returned
+    return set()
```
Comment (yarongilor) [inline sdcm/utils/version_utils.py:231]: is this a valid use-case? would it be usefull at least print a warning about it?
why not simply returning 'None'?

Comment (fruch) [reply sdcm/utils/version_utils.py:231]: there are only 2 options:
https://github.com/scylladb/scylla-cluster-tests/blob/728debeaa5b948f2e9776ced40ea15ae4034c8e4/sdcm/utils/version_utils.py#L70

and as the comment is saying, it's just to please pylint, hence no need to print a warning.

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
@KnifeyMoloko next time, please `git cherry-pick -x` the relevant commits, so it's clearer when comparing between branches, which commits is are in which branch.

**Pull request merged**